src/services/LogFileManager/index.ts

```
@@ -206,22 +235,49 @@ class LogFileManager {
             } due to a limitation in Chromium-based browsers.`);
         }
 
-        return decoder;
+        // Try to match the file extension with a decoder.
+        for (const entry of FILE_TYPE_LIST) {
+            if (entry.extensionList.some((ext) => fileName.endsWith(ext))) {
+                try {
+                    return await entry.Decoder.create(fileData, decoderOptions);
+                } catch (e) {
+                    console.warn(`File extension matches ${entry.typeName},` +
+                        "but decoder creation failed.", e);
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)